Gems/Prefab/PrefabBuilder/PrefabGroup/PrefabGroupBehavior.cpp

```diff
@@ -326,7 +326,7 @@ namespace AZ::SceneAPI::Behaviors
                 meshGroup->GetSceneNodeSelectionList().AddSelectedNode(AZStd::move(meshNodePath));
                 for (const auto& meshGoupNamePair : meshTransformMap)
                 {
-                    if (meshGoupNamePair.first != thisNodeIndex)
+                    if (meshGoupNamePair.second.m_meshIndex != meshNodeIndex)
```

Comment:
nit: A comment would be useful to explain what's going on here for folks who are new to this code.